code/modules/station_goals/bsa.dm

```diff
@@ -353,21 +363,25 @@ CREATION_TEST_IGNORE_SUBTYPES(/obj/machinery/power/bsa/full)
 	if(!ui)
 		ui = new(user, src, "BluespaceArtillery")
 		ui.open()
-		//Missing updates for: target GPS name changes
+		ui.set_autoupdate(TRUE)
 
 /obj/machinery/computer/bsa_control/ui_data()
 	var/obj/machinery/power/bsa/full/cannon = cannon_ref?.resolve()
+	var/datum/component/gps/target = target_ref?.resolve()
 	var/list/data = list()
 	data["ready"] = cannon ? cannon.ready : FALSE
 	data["connected"] = cannon
 	data["notice"] = notice
 	data["unlocked"] = GLOB.bsa_unlock
 	data["charge"] = cannon ? cannon.cell.charge : 0
 	data["max_charge"] = cannon ? cannon.cell.maxcharge : 0
-	if(target)
-		data["target"] = get_target_name()
+	data["formatted_charge"] = cannon ? display_power(cannon.cell.charge) : "0 W"
+	data["targets"] = get_available_targets()
+	if(target_ref?.resolve())
+		data["target"] = list(FAST_REF(target), get_target_name())
 	else
 		data["target"] = null
```

Comment:
```suggestion
	if(target_ref?.resolve())
		data["target_ref"] = FAST_REF(target)
		data["target_name"] = get_target_name()
	else
		data["target_ref"] = null
		data["target_name"] = null
```
instead of sending a list, assign these into two keys

Reply:
i made it like that to have it consistent with targets

Reply:
done